pkg/client/apiutil/restmapper_test.go

```diff
@@ -210,6 +210,28 @@ func TestLazyRestMapperProvider(t *testing.T) {
 		g.Expect(crt.GetRequestCount()).To(gmg.Equal(4))
 	})
 
+	t.Run("LazyRESTMapper KindsFor should work correctly with empty versions list", func(t *testing.T) {
```

Comment:
```suggestion
	t.Run("LazyRESTMapper should work correctly with no versions specified", func(t *testing.T) {
```

Nit

Reply:
There is already a _LazyRESTMapper should work correctly with empty versions list_ test, so I think we should somehow distingush `KindsFor` from `RESTMapping` in the test names 🤔

What about this:
```suggestion
	t.Run("LazyRESTMapper should work correctly with no versions specified in KindsFor", func(t *testing.T) {
```

 and likewise for KindFor, ResourceFor, and ResourcesFor?

Reply:
We could also put all four in the same test

Reply:
Naming the test depends on how you approach https://github.com/kubernetes-sigs/controller-runtime/pull/2346#discussion_r1204573552.

I noticed this "deviation" in naming in the failing test report:

````
--- FAIL: TestLazyRestMapperProvider (6.74s)
    --- PASS: TestLazyRestMapperProvider/LazyRESTMapper_should_fetch_data_based_on_the_request (0.02s)
    --- PASS: TestLazyRestMapperProvider/LazyRESTMapper_should_cache_fetched_data_and_doesn't_perform_any_additional_requests (0.02s)
    --- PASS: TestLazyRestMapperProvider/LazyRESTMapper_should_work_correctly_with_empty_versions_list (0.01s)
    --- FAIL: TestLazyRestMapperProvider/LazyRESTMapper_KindsFor_should_work_correctly_with_empty_versions_list (0.01s)
    --- PASS: TestLazyRestMapperProvider/LazyRESTMapper_should_work_correctly_with_multiple_API_group_versions (0.00s)
    --- PASS: TestLazyRestMapperProvider/LazyRESTMapper_should_work_correctly_with_different_API_group_versions (0.01s)
    --- PASS: TestLazyRestMapperProvider/LazyRESTMapper_should_return_an_error_if_the_group_doesn't_exist (0.01s)
    --- PASS: TestLazyRestMapperProvider/LazyRESTMapper_should_return_an_error_if_a_resource_doesn't_exist (0.03s)
    --- PASS: TestLazyRestMapperProvider/LazyRESTMapper_should_return_an_error_if_the_version_doesn't_exist (0.01s)
    --- PASS: TestLazyRestMapperProvider/LazyRESTMapper_can_fetch_CRDs_if_they_were_created_at_runtime (0.05s)
````

Reply:
Took inspiration from _LazyRESTMapper should return an error if the version doesn't exist_ and called it _LazyRESTMapper should work correctly if the version isn't specified_